dts: gen_defines.py bails out on new path property type

**Describe the bug**

gen_defines.py when provided with a property of type 'path' (introduced in  #21625) bails out with:
```
Traceback (most recent call last):
  File "/zephyr/scripts/dts/gen_defines.py", line 810, in <module>
    main()
  File "/zephyr/scripts/dts/gen_defines.py", line 57, in main
    write_props(node)
  File "/zephyr/scripts/dts/gen_defines.py", line 220, in write_props
    write_phandle_val_list(prop)
  File "/zephyr/scripts/dts/gen_defines.py", line 562, in write_phandle_val_list
    for i, entry in enumerate(prop.val):
TypeError: 'Node' object is not iterable
CMake Error at /zephyr/cmake/dts.cmake:207 (message):
  new extractor failed with return code: 1
Call Stack (most recent call first):
  /zephyr/cmake/app/boilerplate.cmake:460 (include)
  CMakeLists.txt:22 (include)
```

**To Reproduce**
See above.

**Expected behavior**

gen_defines.py to handle 'path' type properties properly.

The assumption in https://github.com/zephyrproject-rtos/zephyr/blob/7cb74655da83bf100eaacc3e00d96c94f1cfbcb3/scripts/dts/gen_defines.py#L219 is not valid anymore. There is now also a 'path' type property.

**Impact**
Annoyance

**Screenshots or console output**
N/A

**Environment (please complete the following information):**
 - Commit 7cb74655da83bf100eaacc3e00d96c94f1cfbcb3


**Additional context**
N/A

--- a/scripts/dts/gen_defines.py
+++ b/scripts/dts/gen_defines.py
@@ -238,8 +238,8 @@ def should_write(prop):
         return False
 
     # For these, Property.val becomes an edtlib.Node, a list of edtlib.Nodes,
-    # and None, respectively. Nothing is generated for them though.
-    if prop.type in {"phandle", "phandles", "compound"}:
+    # or None. Nothing is generated for them at the moment.
+    if prop.type in {"phandle", "phandles", "path", "compound"}:
         return False
 
     # Skip properties that we handle elsewhere

